dvc/tree/local.py

```diff
@@ -326,7 +326,7 @@ def _upload(
         )
 
         self.protect(tmp_file)
-        os.rename(tmp_file, to_info)
+        os.replace(tmp_file, to_info)
```

Comment:
`os.replace` provides the same atomicity guarantee as `os.rename`, now that we check for file existence in `BaseTree.upload`, we should be able to safely default to overwrite behavior in `_upload` across all of our cloud specific trees.